fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvPlanContextBuilder.java

```diff
@@ -19,10 +19,15 @@
 import com.starrocks.qe.ConnectContext;
 
 public class MvPlanContextBuilder {
-    public MvPlanContext getPlanContext(MaterializedView mv) {
+    public static MvPlanContext getPlanContext(MaterializedView mv) {
         // build mv query logical plan
         MaterializedViewOptimizer mvOptimizer = new MaterializedViewOptimizer();
         ConnectContext connectContext = new ConnectContext();
+        // If the caller is not from query (eg. background schema change thread), set thread local info to avoid
+        // NPE in the planning.
+        if (ConnectContext.get() == null) {
+            connectContext.setThreadLocalInfo();
```

Comment:
It only matters when the caller is not from a query and `ConnectContext` is set in the thread local. Because I used it in the SchemaChangeJobV2 which is a background thread, so meets this. 

This should not happen in earlier version?